crates/sp-lightclient/src/lib.rs

```diff
@@ -350,18 +363,15 @@ impl<Header: HeaderT, Store: Storage<Header>> HeaderImporter<Header, Store> {
         ancestor_number: NumberOf<Header>,
     ) -> Option<HeaderExt<Header>> {
         // header number must be greater than the ancestor number
-        if ancestor_number >= *header.header.number() {
+        if *header.header.number() <= ancestor_number {
             return None;
         }
 
         let headers_at_ancestor_number = self.store.headers_at_number(ancestor_number);
         let finalized_header = self.store.finalized_header();
 
         // short circuit if the ancestor number is at the same or lower number than finalized head
-        if ancestor_number.le(finalized_header.header.number())
-            // short circuit if there are no forks at the depth
-            || headers_at_ancestor_number.len() == 1
-        {
+        if ancestor_number <= *finalized_header.header.number() {
```

Comment:
I meant to simplify to `headers_at_ancestor_number.len() == 1`, which is more generic case than `ancestor_number <= *finalized_header.header.number()` and works in more cases where block is not finalized yet.